compiler/rustc_resolve/src/lib.rs

```diff
@@ -2523,19 +2523,29 @@ impl<'a> Resolver<'a> {
                         } else {
                             (
                                 format!("use of undeclared crate or module `{}`", ident),
-                                self.find_similarly_named_module_or_crate(
-                                    ident.name,
-                                    &parent_scope.module,
-                                )
-                                .map(|sugg| {
-                                    (
-                                        vec![(ident.span, sugg.to_string())],
+                                if ident.name == sym::alloc {
+                                    Some((
+                                        vec![],
                                         String::from(
-                                            "there is a crate or module with a similar name",
+                                            "add `extern crate alloc` to use the builtin `alloc` module",
```

Comment:
```suggestion
                                            "add `extern crate alloc` to use the `alloc` crate",
```
I'm not sure if "builtin" is accurate since `alloc` is separate from the compiler (as far as I'm aware, at least). Alternative phrasings (aside from the one suggested here) might include "the standard `alloc` crate" or "the `alloc` crate in the sysroot".

Reply:
Thank you. I applied your suggestion.